dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/spirv/SpirvCmpXchg.java

```diff
@@ -0,0 +1,78 @@
+package com.dat3m.dartagnan.program.event.lang.spirv;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.EventVisitor;
+import com.dat3m.dartagnan.program.event.common.RMWCmpXchgBase;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.dat3m.dartagnan.program.event.Tag.Spirv.*;
+
+public class SpirvCmpXchg extends RMWCmpXchgBase {
+
+    private static final List<String> moStrength = List.of(RELAXED, RELEASE, ACQUIRE, ACQ_REL, SEQ_CST);
+
+    private final String scope;
+    private final String eqMo;
+    private final Set<String> eqTags;
+
+    public SpirvCmpXchg(Register register, Expression address, Expression cmp, Expression value, String scope, Set<String> eqTags, Set<String> neqTags) {
+        super(register, address, cmp, value, true, getMoTag(neqTags));
```

Comment:
See the comment above. SpirvCmpXchg should correctly represent the corresponding spirv operation. If the combination of tags is such that we don't support, then it will fail compilation to vulkan.